### PR TITLE
Machine Learning image classification with Javascript and Nyckel

### LINKED ISSUE #4833: [Machine Learning] Machine Learning Image Classification with Javascript and Nyckel
#### Introduction

Artificial intelligence has become the future of technology and everyone is trying to apply it in all areas. Artificially intelligent applications are a huge contribution to AI technology with the likes of trading bots and others.

This article shows how you can implement machine learning models in a web application to classify images. In this case, we will classify and identify the images of people that are wearing and those who are not wearing a mask.

#### Key takeaways 
- Make an image classification model
- Develop a simple web app
- Integrate the classification model into the application
- Test the application(AI-based)
- Evaluate the application performance.

#### Article quality
This article is intended to show how machine learning can be implemented in web applications, unlike other articles that only teach how to make machine learning models.
#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by vifabulous**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4833 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by vifabulous** [comment]
@jumamiller You can have a look at the updates I made on the article

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        }
+        reader.readAsDataURL(file);
+    }
+    else
+    {
+        alert("You must choose an image");
+        resetPage();
+    }
+}
+```
```
Comment (jumamiller) [inline machine-learning-image-classification-with-javascript-and-nyckel/index.md:201]: where is the explanation for all these?

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+    for (var i = 0; i < rawLength; ++i)
+    {
+        uInt8Array[i] = raw.charCodeAt(i);
+    }
+
+    return new Blob([uInt8Array], { type: contentType });
+}
+
+```
```
Comment (jumamiller) [inline machine-learning-image-classification-with-javascript-and-nyckel/index.md:235]: Where is the expanation?

**Review by jumamiller — CHANGES_REQUESTED** [review]
Did you go thro' my comments on why the previous PR was closed? A tutorial is not all about writing code, it's about walking the reader thro what you're doing. For example, you define `displayResult(response)`, tell the reader why this method is being used in the code, it's that simple. 

Kindly refine the article! Thank you

**Comment by vifabulous** [comment]
@jumamiller I have divided the code into several parts where I have explained the purpose of those functions before each code. The paragraphs I have added explain the purpose of every part....take a look

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md (hunk 1 above)
Comment (vifabulous) [reply machine-learning-image-classification-with-javascript-and-nyckel/index.md:201]: Right before the code I have explained what it does...To make it simpler, the function names I have used makes sure that the reader identifies the purpose of each.

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md (hunk 2 above)
Comment (vifabulous) [reply machine-learning-image-classification-with-javascript-and-nyckel/index.md:235]: I have written an explanation about canvas and blob... and indicated that the code is to convert canvas to blob

**Comment by vifabulous** [comment]
@jumamiller  Seems I had not uploaded the actual file..... I have added more explanations as comments to the updated one, I believe everyone can follow clearly as it is.

**Comment by Kayere** [comment]
Hello @vifabulous

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Machine Learning image classification with Javascript and Nyckel #5411 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7692909/Machine.Learning.image.classification.with.Javascript.and.Nyckel.5411.Copyleaks.Report.pdf)

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+    }
+    else
+    {
+    // If the file is not an image, upload is rejected with an alert and the page is reset to default
+        alert("You must choose an image");
+        resetPage();
+    }
+}
+```
+The rectangular area for hosting the uploaded image is called a **canvas** in html. A **blob** on the other hand represents objects in form of raw data. Converting canvas to blob translates the images uploaded to your application in a way that the model can understand.
```
Comment (jumamiller) [inline machine-learning-image-classification-with-javascript-and-nyckel/index.md:209]: Hi, could you show me the explanations for this code snippet? (Not comments!)

**Review by jumamiller — COMMENTED** [review]
Just a quick one, I have gone thro' your article and I cant trace the code explanations, maybe you point them out to me if possible!

FYI: The main reason I closed the previous PR was for the same reason but I can't see improvements!

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md (hunk 3 above)
Comment (vifabulous) [reply machine-learning-image-classification-with-javascript-and-nyckel/index.md:209]: @jumamiller this paragraph `The rectangular area for hosting the uploaded image is called a **canvas** in html. A **blob** on the other hand represents objects in form of raw data. Converting canvas to blob translates the images uploaded to your application in a way that the model can understand` gives the general objective of the code while the many comments I added gives the code explanation/objective where needed. Kindly check them out

Comment (vifabulous) [reply machine-learning-image-classification-with-javascript-and-nyckel/index.md:209]: Please reconsider the comments

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+This code is placed under the `<head>` tag creates a connection to the JavaScript file.
+```html
+<html>
+
+<head>
+  <script src="https://code.jquery.com/jquery-latest.min.js"></script>
+</head>
+
+```
+
```
Comment (jumamiller) [inline machine-learning-image-classification-with-javascript-and-nyckel/index.md:43]: Hi, could you read this section slowly as if you were a reader!

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+Under the body, the tag is where the upload button and the upload section lies. This code creates what is required under the body section without any styling.
```
Comment (jumamiller) [inline machine-learning-image-classification-with-javascript-and-nyckel/index.md:44]: This too, does it add up?

File: machine-learning-image-classification-with-javascript-and-nyckel/index.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+    <div class="upload-area" id="uploadfile">
+      <h2>Drop Your photo here</h2>
+    </div>
+    <input type="file" name="file" id="file">
+    <a id="nyckel" href="https://www.nyckel.com">Machine Learning application created by Victor</a>
+  </div>
+</body>
+```
+
+Once uploaded, a container is created to hold the photo using the division class `upload-area`.
```
Comment (jumamiller) [inline machine-learning-image-classification-with-javascript-and-nyckel/index.md:62]: Once uploaded? what's being uploaded here while you're describing how to build...

**Review by jumamiller — COMMENTED** [review]
I really feel the quality of this article is extremely low given it's a duplicate.

**Comment by jumamiller** [comment]
Finalising the review!

**Comment by vifabulous** [comment]
I have made the required changes @jumamiller..... I will try to have little to no mistakes in my future writings

**Review by jumamiller — APPROVED** [review]

**Pull request merged**